flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java

```diff
@@ -141,6 +140,22 @@ protected Connection setupConnection() throws Exception {
 		return setupConnectionFactory().newConnection();
 	}
 
+	/**
+	 * Initializes the consumer's {@link Channel}. If a prefetch count has been set in {@link RMQConnectionConfig},
+	 * the new channel will be use it for {@link Channel#basicQos(int)}.
+	 *
+	 * @param connection the consumer's {@link Connection}.
+	 * @return the channel.
+	 * @throws Exception if there is an issue creating or configuring the channel.
+	 */
+	protected Channel setupChannel(Connection connection) throws Exception {
+		Channel chan = connection.createChannel();
+		if (rmqConnectionConfig.getPrefetchCount().isPresent()) {
+			chan.basicQos(rmqConnectionConfig.getPrefetchCount().get());
```

Comment:
should we set the global flag to true ? to enforce having only that limit throughout the channel ? 
i am worried if someone adds a functionality that say requires having multiple consumers on the same channel that then the `preFetchCount` would be per consumer and the preFetchCount set by the client would suddenly mean a completely different thing. But if enforced here to be per channel it will have the same effect for the user.

According to the documentation  `chan.basicQos(prefectchCount, boolean global)` and the boolean `global` is: 

```
false | applied separately to each new consumer on the channel
true  | shared across all consumers on the channel
```

Reply:
Thanks for going through this! That's a good point, I'm not sure what a good answer is. I've read in the Nodejs client I used that [sending that flag to a RMQ cluster < version 3.3.0 will kill it entirely](https://www.squaremobius.net/amqp.node/channel_api.html#channel_prefetch), but I haven't tested that out at all or seen it elsewhere. I'll put together a playground this week. 

We're always reading off of one queue per source in our jobs, but our use case is pretty simple so I'm not sure I've got the best understanding of other RMQ users. What kinds of reasons do people have for sharing a channel?

Reply:
Hemm ... you're absolutely right i also have a very basic setup several independent brokers and we duplicate and deduplicate on the clients.

`To keep life interesting, using the global flag with an RabbitMQ older than v3.3.0 will bring down the whole connection.` :rofl: okay then experimenting is the way to go !  do you need a helping hand ? i can spare a couple of hours next weekend if you need help.

Reply:
I'll let you know how it goes and thank you for the offer - I'll take you up on it if I can't find time this week! Moving this weekend, so it'll be busy over here..

Reply:
Finally got around to this, sorry about the late reply! When I was putting together the example, I found that RMQ 3.3 is not even available on Docker hub and has [been out of service since 2015](https://www.rabbitmq.com/versions.html) and [the last patch was released in 2016](https://github.com/rabbitmq/rabbitmq-server/releases/tag/rabbitmq_v3_3_5), so I think we can safely just set the global flag and not care about that version. Here's the playground running this branch's code: https://github.com/austince/flink-rmq-playground